pkg/async/async.go

```diff
@@ -114,6 +121,44 @@ func RunBackground(ctx context.Context, r Runner) {
 	Run(context.Background(), r)
 }
 
+// RunClose closes any references a runner might be using
+func RunClose(ctx context.Context, r Runner) error {
+	switch r := r.(type) {
+	case Closer:
+		return r.Close(ctx)
+	case io.Closer:
+		return r.Close()
+	}
+	return nil
+}
+
+// RunGroup runs a group of runner tasks and exits when the first run group errors out
+func RunGroup(rg []Runner) Runner {
+	ru := Func(func(ctx context.Context) error {
+		g, ctx := errgroup.WithContext(ctx)
+		ctx, cancelFunc := context.WithCancel(ctx)
+		defer cancelFunc()
+		for idx := range rg {
+			r := rg[idx]
+			g.Go(func() error {
+				defer func() {
+					if err := RunClose(ctx, r); err != nil {
+						log.Error(ctx, "Error when closing:", events.NewErrorInfo(err))
+					}
+				}()
+
+				err := r.Run(ctx)
+				if err != nil {
+					cancelFunc()
```

Comment:
lol, this is exactly what `errgroup` itself does internally: see [here](https://cs.opensource.google/go/x/sync/+/036812b2:errgroup/errgroup.go;l=61) -- so you really do not need to do anything other than just `return r.Run(ctx)`

Reply:
I guess I got confused by the comment on the wait method 
“ Wait blocks until all function calls from the Go method have returned, then
// returns the first non-nil error (if any) from them” 
Just assumed that meant it’s blocking till all the tasks finished

Reply:
Learn something new everyday I guess. Thanks  for the help with this.

Reply:
Yeah, technically it does wait till all complete but it does cancel all the go-routines.  This is the same behavior in our current service activities I think

Reply:
Yep it is. So maybe we could replace the service activities code with this now.... most of the pieces are here now ... with the closer & RunGroup, only thing left is to pass in the Config as part of the run method.

Reply:
Yeah we can switch to this once we change the service activities to take the `config` via the struct initialization rather than as argument to `Run`.